### PR TITLE
Fix default quoteEscapeCharacter

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Fix default quoteEscapeCharacter

## Motivation and Context
QuoteEscapeCharacter is `\\` not '"` 

## Regression
Yes perhaps during select refactors

## How Has This Been Tested?
Using the spark-select project with latest master and 

```scala
import org.apache.spark.sql._
import org.apache.spark.sql.types._

object app {
  def main(args: Array[String]) {
    val schema = StructType(
      List(
        StructField("name", StringType, true),
        StructField("age", IntegerType, false)
      )
    )

    val df = spark
      .read
      .format("minioSelectCSV")
      .schema(schema)
      .load("s3://sjm-airlines/people.csv")

    println(df.show())

    println(df.select("*").filter("age > 19").show())

  }
}
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

File: pkg/s3select/csv/args.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ const (
 	defaultRecordDelimiter      = "\n"
 	defaultFieldDelimiter       = ","
 	defaultQuoteCharacter       = `"`
-	defaultQuoteEscapeCharacter = `"`
+	defaultQuoteEscapeCharacter = `\\`
```
Comment (balamurugana) [inline pkg/s3select/csv/args.go:33]: As per csv, `"` is default quote escape character.  Have we decided to change it?

Comment (donatello) [reply pkg/s3select/csv/args.go:33]: Refer https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectSELECTContent.html

**Review by donatello — CHANGES_REQUESTED** [review]

File: pkg/s3select/csv/args.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/s3select/csv/args.go]: We need to keep it like AWS S3 not like CSV

Comment (harshavardhana) [reply pkg/s3select/csv/args.go:33]: The issue is perhaps we don't support anything other than `"` quote escape character

**Comment by harshavardhana** [comment]
Looks like the document from EMR select vs select object are not in sync, will close this PR for now.

**Pull request closed without merging**